src/cow-react/modules/limitOrders/pure/DeadlineSelector/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { Dropdown } from '@cow/common/pure/Dropdown'
+import { LimitOrderDeadline, limitOrdersDeadlines, maxCustomDeadline } from './deadlines'
+
+import { useCallback, useMemo, useRef } from 'react'
+import { ChevronDown } from 'react-feather'
+import * as styledEl from './styled'
+import ms from 'ms.macro'
+import { Trans } from '@lingui/macro'
+
+function limitDateString(date: Date): string {
+  const [first, second] = date.toISOString().split(':')
+
+  return [first, second].join(':')
+}
+
+const customDateOptions: Intl.DateTimeFormatOptions = {
+  year: '2-digit',
+  month: 'short',
+  day: 'numeric',
+  hour: '2-digit',
+  minute: '2-digit',
+  hour12: false,
+}
+
+export interface DeadlineSelectorProps {
+  deadline: LimitOrderDeadline | undefined
+  customDeadline: number | null
+  selectDeadline(deadline: LimitOrderDeadline): void
+  selectCustomDeadline(deadline: number): void
+}
+
+export function DeadlineSelector(props: DeadlineSelectorProps) {
+  const { deadline, customDeadline, selectDeadline, selectCustomDeadline } = props
+  const currentDeadlineNode = useRef<HTMLButtonElement>()
+
+  const min = limitDateString(new Date(Date.now() + ms`10m`))
+  const max = limitDateString(new Date(Date.now() + maxCustomDeadline))
+
+  const existingDeadline = useMemo(() => {
+    return limitOrdersDeadlines.find((item) => item === deadline)
+  }, [deadline])
+
+  const customDeadlineTitle = useMemo(() => {
+    if (!customDeadline) return ''
+    return new Date(customDeadline).toLocaleString(undefined, customDateOptions)
+  }, [customDeadline])
+
+  const setDeadline = useCallback(
+    (deadline: LimitOrderDeadline) => {
+      selectDeadline(deadline)
+      currentDeadlineNode.current?.click() // Close dropdown
+    },
+    [selectDeadline]
+  )
+
+  const onChange = useCallback(
+    (event) => {
+      const customDeadline = new Date(event.target.value).getTime()
+
+      selectCustomDeadline(customDeadline)
+    },
+    [selectCustomDeadline]
+  )
+
+  const list = (
+    <styledEl.ListWrapper>
+      {limitOrdersDeadlines.map((item) => (
+        <li key={item.value}>
+          <styledEl.ListItem onClick={() => setDeadline(item)}>
+            <Trans>{item.title}</Trans>
+          </styledEl.ListItem>
+        </li>
+      ))}
+      <styledEl.ListItem>
+        <Trans>Custom</Trans>
+        <styledEl.CustomInput type="datetime-local" onChange={onChange} min={min} max={max} />
+      </styledEl.ListItem>
+    </styledEl.ListWrapper>
+  )
+
+  return (
+    <styledEl.Wrapper>
+      <styledEl.Title>
+        <Trans>Expiry</Trans>
+      </styledEl.Title>
+      <Dropdown content={list}>
+        <styledEl.Current ref={currentDeadlineNode as any} isCustom={!!customDeadline}>
```

Comment:
At first i thought the custom datetimepicker was not fitting to well, but I saw you use the native, so I see that having some advantages.

I thought we would make something custom to make it look exactly how we want, but I kind of like your approach to adapt to the native device. Plus, I don't think people will use custom too much.

Reply:
@michelbio do you think is ok we leave the "custom" using the browser native datepicker?

Reply:
As far as I can see in this PR, it still uses a custom dropdown and not a browser native dropdown?

In general I see some benefits to use a native option, especially for mobile devices (iOS, Android).

Reply:
You can test here: 
https://pr1283--cowswap.review.gnosisdev.com/#/1/limit-orders/WETH

<img width="941" alt="image" src="https://user-images.githubusercontent.com/2352112/200034940-cd4425ea-17d8-45c1-ab95-ef975be22ae8.png">

<img width="1104" alt="image" src="https://user-images.githubusercontent.com/2352112/200035151-038aa7ba-46c3-448e-a7c6-36c6c38920e2.png">

Reply:
>Plus, I don't think people will use custom too much.
I think so